src/processor/operator/index_lookup.cpp

```diff
@@ -63,60 +66,59 @@ void IndexLookup::fillOffsetArraysFromVector(transaction::Transaction* transacti
     KU_ASSERT(resultVector->dataType.getPhysicalType() == PhysicalTypeID::INT64);
     auto offsets = (offset_t*)resultVector->getData();
     auto numKeys = keyVector->state->selVector->selectedSize;
-    switch (info.pkDataType->getLogicalTypeID()) {
-    case LogicalTypeID::INT64: {
-        if (info.copyNodeSharedState == nullptr) {
-            for (auto i = 0u; i < numKeys; i++) {
-                auto pos = keyVector->state->selVector->selectedPositions[i];
-                auto key = keyVector->getValue<int64_t>(pos);
-                if (!info.index->lookup(transaction, key, offsets[i])) {
-                    throw RuntimeException(
-                        ExceptionMessage::nonExistentPKException(std::to_string(key)));
+    TypeUtils::visit(
+        info.pkDataType->getPhysicalType(),
+        [&](ku_string_t) {
```

Comment:
Can we move these two lambdas for `ku_string_t` and primitive types to functions? That is more readable.